website/scripts/index_search_content.js

```diff
@@ -1,3 +1,81 @@
-const { indexDocsContent } = require('@hashicorp/react-search/tools')
+require('dotenv').config()
+const fs = require('fs')
+const path = require('path')
+const {
+  indexContent,
+  getDocsSearchObject,
+} = require('@hashicorp/react-search/tools')
+const resolveNavData = require('../components/remote-plugin-docs/utils/resolve-nav-data')
+const fetchGithubFile = require('../components/remote-plugin-docs/utils/fetch-github-file')
 
-indexDocsContent()
+// Run indexing
+indexContent({ getSearchObjects })
+
+async function getSearchObjects() {
+  // Resolve /docs, /guides, and /intro nav data, which
+  // corresponds to all the content we will actually render
+  // (this avoids indexing non-rendered content, and partials)
+  // `docs` content
+  const docsNav = await resolveNavData(
+    'data/docs-nav-data.json',
+    'content/docs',
+    { remotePluginsFile: 'data/docs-remote-plugins.json' }
+  )
+  const docsObjects = await searchObjectsFromNavData(docsNav, 'docs')
+  // `guides` content
+  const guidesNav = await resolveNavData(
+    'data/guides-nav-data.json',
+    'content/guides'
+  )
+  const guidesObjects = await searchObjectsFromNavData(guidesNav, 'guides')
+  // `intro` content
+  const introNav = await resolveNavData(
+    'data/intro-nav-data.json',
+    'content/intro'
+  )
+  const introObjects = await searchObjectsFromNavData(introNav, 'intro')
+  // Concat all search objects, and return them
+  const searchObjects = [].concat(docsObjects, guidesObjects, introObjects)
```

Comment:
nit: this fn might read a little easier if you define this as an empty array at the top, and then merge each successive set of content into it. Initially when I got to L19 I thought "wait, where are the search objects? 🤔"

```js
const searchObjects = []

const docsNav = await resolveNavData(/* ... */)

searchObjects.push(...docsNav)
// etc

Reply:
Sweet, I like this better - have updated to match 👍